src/projectManager.ts

```diff
@@ -68,15 +70,42 @@ export class ProjectManager {
         return;
     }
 
+    public static getProjects(): IProject[] {
+        let projects = [];
+        for (const index in this.loaded) {
+            projects.push(this.loaded[index]);
+        }
+
+        return projects;
+    }
+
+    public static getProjectFromActiveTextEditor(): IProject | undefined {
+        let activeFileUri = window.activeTextEditor?.document.uri;
+        activeFileUri = activeFileUri?.scheme === 'file' ? activeFileUri : undefined;
+
+        if (activeFileUri) {
+            return this.getProjectFromUri(activeFileUri);
+        }
+    }
+
     public static getProjectFromUri(uri: Uri): IProject | undefined {
-        const workspaceFolders = workspace.workspaceFolders;
-        if (workspaceFolders && workspaceFolders.length > 0) {
-            const changedWorkspaceFolder = workspaceFolders.filter(workspaceFolder =>
-                uri.fsPath.startsWith(workspaceFolder.uri.fsPath)
-            )[0];
+        for (const index in this.loaded) {
```

Comment:
This for loop can be replaced by a simple `workspace.getWorkspaceFolder` API call, which returns `WorkspaceFolder`, which has the project index on it.

Reply:
Good catch. Updated this